.github/workflows/CI.yml

```diff
@@ -0,0 +1,193 @@
+name: CI
+on:
+  # always execute docker build when something is pushed to main/master or release-* branches
+  push:
+    branches:
+      - "master"
+      - "main"
+      - "release-*"
+  # in addition, execute for pull requests to those branches
+  pull_request:
+    branches:
+      - "master"
+      - "main"
+      - "release-*"
+  # run integration tests when triggered manually
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+jobs:
+  prepare_ci_run:
+    name: Prepare CI Run
+    # Prepare CI Run looks at what has been changed in this commit/PR/... and determines which artifacts should be
+    # built afterwards (in other jobs that depend on this one).
```

Comment:
Is this really the case?

Reply:
The CI workflow is from the keptn-service-template-go: https://github.com/keptn-sandbox/keptn-service-template-go/blob/main/.github/workflows/CI.yml#L23-L25.

Reply:
yes, but it still doesn't check for changed files and also doesn't determine if something should be built from what I can tell